tensorflow/compiler/tf2tensorrt/kernels/trt_engine_resource_ops_test.cc

```diff
@@ -223,6 +337,35 @@ TEST_F(TRTEngineResourceOpsTest, Basic) {
   // the cache of the resource is not empty.
   EXPECT_TRUE(rm->Lookup(container, resource_name, &resource).ok());
   EXPECT_EQ(1, resource->cache_.size());
+  if (this->param_.dynamic) {
+    EXPECT_EQ(3, resource->profiles_.GetNumProfiles());
+    EXPECT_EQ(3, resource->cache_.begin()->second->execution_context.size());
+
+    if (this->param_.network == 1) {
+      // Check if profiles are restored correctly.
+      std::vector<TensorShape> shapes(1);
+      // We create a shape vector that matches only profile 1.
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{6}, &shapes[0]));
+      EXPECT_EQ(1, resource->profiles_.GetProfileNumber(shapes));
+    } else {
+      // Check if shape values are restored corretly.
+      std::vector<TensorShape> shapes(2);
+      // We create a shape vector that matches only profile 2.
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{9, 9}, &shapes[0]));
+      TF_CHECK_OK(
+          TensorShapeUtils::MakeShape(std::vector<int32>{2}, &shapes[1]));
+      Tensor shape_tensor(DT_INT32, shapes[1]);
+      std::vector<int32> vals{1, 3};
+      std::copy_n(vals.data(), vals.size(),
+                  shape_tensor.flat<int32_t>().data());
+      DataVec shape_values{{"one", {}}, {"two", shape_tensor}};
+      // DataVec names are not in CollectShapeValues, used only order matters.
```

Comment:
Sorry, I don't understand the second part of this comment "used only order matters". Can you explain?
Should this comment be moved to before shape_values?